### DISCUSSION TITLE
screws_tilt_adjust

### DISCUSSION THREAD

**Discussion by Arnaud3767** [xblax/flashforge_ad5m_klipper_mod#197] [Ideas]
Hello,
I'm not good at configuring klipper yet.
Is it possible to help me create screws_tilt_adjust to best adjust the screws on my bed?

klipper3D.org exemple:

[screws_tilt_adjust]
screw1: -5, 30
screw1_name: front left screw
screw2: 155, 30
screw2_name: front right screw
screw3: 155, 190
screw3_name: rear right screw
screw4: -5, 190
screw4_name: rear left screw
horizontal_move_z: 10.
speed: 50.
screw_thread: CW-M3

**Comment by b-fitzpatrick** [comment]
Here's what I use:
```
[screws_tilt_adjust]
screw1: -94.5, -94.5
screw1_name: front-left
screw2: 94.5, -94.5
screw2_name: front-right
screw3: 94.5, 94.5
screw3_name: rear-right
screw4: -94.5, 94.5
screw4_name: rear-left
horizontal_move_z: 2
speed: 200
screw_thread: CW-M4
```

**Reply by Arnaud3767** [reply]
Great, thank you

**Comment by caall99** [comment]
There is only one motor for z, how can you tilt adjust the bed on the Adventurer 5?

**Reply by b-fitzpatrick** [reply]
There are four spring-loaded screws that mount the bed. They're accessible when you remove the build plate. You also need to hold a lock nut underneath when adjusting.

There may be four additional screws in the middle of each side, but they have been omitted in two of my three printers. They made things worse in the one that has them, so I loosened them and will probably remove them.

My best one:
![image](https://github.com/xblax/flashforge_ad5m_klipper_mod/assets/9221473/31e9ff66-c794-4b51-94dd-43eefb5386f0)

And my worst one: (would like to figure out how to improve this)
![image](https://github.com/xblax/flashforge_ad5m_klipper_mod/assets/9221473/92bb25f9-3d2e-4247-9d89-0b803af748a1)

**Reply by caall99** [reply]
Maybe I am confusing it with z tilt...

**Reply by b-fitzpatrick** [reply]
Yeah, not like rotating the three z screws independently, but worthwhile if you care about a level bed (for functional part purposes, not for first-layer quality).

**Comment by caall99** [comment]
This manual leveling procedure is one of the first things I did when I purchased my printer to get a more even and level bed mesh. I just did it extremely manually, by moving the toolhead myself and then adjusting the screws... lol! Didn't realize klipper had a routine for that!

**Comment by thhdragon** [comment]
i actually did this too recently. also added in redoing the 4 other screws because i think my top one was cranked down too hard and causing my bed or something to bind because there was a lump on my mesh. im not a fan of probing once per spot i use this code 
```
samples: 3
samples_result: median
sample_retract_dist: 3.0
samples_tolerance: 0.01
samples_tolerance_retries: 6
```
added to my probe section. before i literally had to set my samples tolerance to .05, now im using .01 and it usually only retries a few times
![Screenshot 2024-06-09 052639](https://github.com/xblax/flashforge_ad5m_klipper_mod/assets/46762155/790a7d47-efa6-4857-a5e0-02e7c9c50a52)
old mesh^^
![image](https://github.com/xblax/flashforge_ad5m_klipper_mod/assets/46762155/872f847f-bf0d-4e17-87b6-1a5239984e6b)
^^new mesh and at 0.01 probe tolerance

**Comment by Dedall** [comment]
Hi. Maybe somebody knows how to fix my bed. I calibrated with screws_tilt_adjust but I still have 0.5
![448406017_818339220399837_7376475972343972275_n](https://github.com/xblax/flashforge_ad5m_klipper_mod/assets/56481276/2ec99f41-000c-42f0-abbd-152282441357)

**Reply by b-fitzpatrick** [reply]
That looks similar to my worst one above. The bed is slightly warped. I've thought about trying to build up the low areas with aluminum tape or just trying a replacement bed.

**Reply by thhdragon** [reply]
the 4 screws in the center cross (not the ones you used to level) are self tapping screws that screw into that plastic part. i took them out and put them back in making sure not to tighten them too much. if they are too tight they can cause the bed to be pulled down at that spot
